### PR TITLE
Add support for passing path to custom TLS CA cert & client key/cert …

### LINKED ISSUE #184: Support Alertmanager with TLS Auth?
Dear all,

I have many Alertmanager and somes with TLS auth. 
Is it possible to use a client cert in Unsee to scrape alerts?

Regards

### PR DESCRIPTION
**Pull request by prymitive**
…pair

Fixes #184

### CODE REVIEW

File: internal/alertmanager/models.go [hunk 1]
```diff
@@ -92,7 +101,7 @@ func (am *Alertmanager) pullSilences(version string) error {
 	}
 
 	start := time.Now()
-	silences, err := mapper.GetSilences(am.URI, am.RequestTimeout)
+	silences, err := mapper.GetSilences(am.URI, am.RequestTimeout, am.TLSConfig)
```
Comment (terinjokes) [inline internal/alertmanager/models.go]: Why are you having to thread TLSConfig? I'd expect the transport to be allocated (with the TLS config) when the AM was added.

Comment (prymitive) [reply internal/alertmanager/models.go]: yeah, that's not great, refactored to pass around tls.Config instead
anything else stands out here?

File: internal/alertmanager/upstream.go [hunk 2]
```diff
@@ -82,15 +87,31 @@ func GetAlertmanagerByName(name string) *Alertmanager {
...
+	return func(am *Alertmanager) error {
 		am.RequestTimeout = timeout
+		return nil
+	}
+}
+
+// WithTLSConfig option can be passed to NewAlertmanager in order to set
+// custom options for the TLS handling when sending requests to Alertmanager
+// upstream
+func WithTLSConfig(tlsOptions transport.TLSConnectionOptions) Option {
```
Comment (terinjokes) [inline internal/alertmanager/upstream.go]: Either rename this function to `WithTLSOptions` or similar or do this work to turn the YAML options into a standard `tls.Config` and pass that here instead.

Comment (prymitive) [reply internal/alertmanager/upstream.go]: I think I'm gonna go with dedicated option for setting CA certs and client cert/key

**Comment by prymitive** [comment]
@terinjokes  please review latest version, with previous refactoring this is now a smaller
there's some extra noise due to renaming of `transport` package, I felt it was needed since `http.Transport` is now passed (rather than just `tls.Config`) and having 2 "transports" passed around was too confusing

**Pull request merged**